src/prog_models/sim_result.py

```diff
@@ -34,6 +34,32 @@ def __init__(self, times: list = None, data: list = None, _copy=True):
             else:
                 self.data = data
 
+    def __getitem__(self, item):
+        """
+        created for deprecation warning. [] continues to be handled by parent
+        """
+        warn(
+            '[] will be deprecated after version 1.5 of ProgPy. Users may use pandas Dataframe \'iloc\'  to access '
+            'data by element.',
+            DeprecationWarning, stacklevel=2)
+        return super()[item]
+    
+    def __iter__(self):
+        """
+        created for deprecation warning. iteration continues to be handled by parent
+        """
+        warn(
+            'iteration will be deprecated after version 1.5 of ProgPy. The function will be renamed, iterrows, '
+            'and users may begin using it under this name now.',
+            DeprecationWarning, stacklevel=2)
+        super().__iter__(self)
+    
+    def iterrows(self):
+        """
+            Iterates -- through keys
+        """
+        super().__iter__(self)
```

Comment:
I actually looked at iterrows, and I think we want to call it itertuples. Iterrows return the index. 

You can also include iterrows and have it return enumerate(self.data)